src/xform.c

```diff
@@ -1142,12 +1151,56 @@ resize_stream(Gif_Stream* gfs,
         && method != SCALE_METHOD_LANCZOS3 && method != SCALE_METHOD_MITCHELL)
         method = SCALE_METHOD_POINT;
 
-    for (sctx.imageno = 0; sctx.imageno < gfs->nimages; ++sctx.imageno) {
-        sctx.gfi = gfs->images[sctx.imageno];
-        scale_image(&sctx, method);
+#ifdef ENABLE_THREADED
+    pthread_mutex_init(&kd3_sort_lock, 0);
+#endif
+
+    int i;
+    if (resize_threads > 0) {
+#ifdef ENABLE_THREADED
+        scale_thread_context* contexts = malloc(gfs->nimages * sizeof(scale_thread_context));
+        pthread_t * pthreads = malloc(gfs->nimages * sizeof(pthread_t));
+
+        for (i = 0; i < gfs->nimages;) {
+            int start = i;
+            int end = min(i + resize_threads, gfs->nimages);
+            for (; i < end; ++i) {
+                sctx_init(&(contexts[i].sctx), gfs, nw, nh);
+                contexts[i].sctx.imageno = i;
+                contexts[i].sctx.scale_colors = scale_colors;
+                contexts[i].sctx.gfi = gfs->images[i];
+                contexts[i].method = method;
+
+                /* Create worker thread */
+                pthread_create(&pthreads[i], NULL, scale_image_threaded, &contexts[i]);
+            }
+            int j = start;
+            for (; j < end; ++j) {
+                pthread_join(pthreads[j], NULL);
+            }
+        }
```

Comment:
(It's a bit of a shame that __builtin_add_and_fetch is not necessarily available on all compilers; we can configurize around that.)